.github/workflows/elixir.yml

```diff
@@ -115,22 +139,53 @@ jobs:
         run: mix compile
       # Don't cache PLTs based on mix.lock hash, as Dialyzer can incrementally update even old ones
       # Cache key based on Elixir & Erlang version (also useful when running in matrix)
-      - name: Restore PLT cache
+      - uses: actions/cache/restore@v3
+        name: Restore PLT cache
         id: plt_cache
-        uses: actions/cache@v3
         env:
           cache-name: cache-erlang-plt-${{ env.MIX_ENV }}
         with:
-          key: ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-${{ steps.setup-beam.outputs.otp-version }}-plt
+          key:
+            ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-plt
           restore-keys: |
-            ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-${{ steps.setup-beam.outputs.otp-version }}-plt
+            ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-plt
           path: elixir/priv/plts
       - name: Create PLTs
         if: steps.plt_cache.outputs.cache-hit != 'true'
         run: mix dialyzer --plt
       - name: Run Dialyzer
         run: mix dialyzer --format dialyxir
-  elixir_static-analysis:
+      - uses: actions/cache/save@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
+        name: Save Elixir Deps Cache
+        env:
+          cache-name: cache-elixir-deps-${{ env.MIX_ENV }}
+        with:
+          path: elixir/deps
+          key:
+            ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-${{
+            env.cache-name }}-${{ hashFiles('**/elixir/mix.lock') }}
+      - uses: actions/cache/save@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
+        name: Save Elixir Build Cache
+        env:
+          cache-name: cache-elixir-build-${{ env.MIX_ENV }}
+        with:
+          path: elixir/_build
+          key:
+            ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-${{
+            env.cache-name }}-${{ hashFiles('**/elixir/mix.lock') }}
+      - uses: actions/cache/save@v3
+        if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Let's save Elixir cached on all branches and then pull `main` one if branch doesn't have it's own? Elixir cache is rather small but allows for much better compilation times, especially PLT ones.

Reply:
Kk. Let's get this merged so can get a cache saved on `main`, then I'll open another PR exploring this change 👍 if that works